File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        vector <int> ans;
        // following hint 3
        for (int i = 0; i < boxes.size(); i++) {
            int temp = 0;
            for (int j = 0; j < boxes.size(); j++) {
                if (boxes[j] == '1') {
                    temp += abs(i - j);
                }
            }
            ans.push_back(temp);
        }
        return ans;
    }
};

File: 1769-minimum-number-of-operations-to-move-all-balls-to-each-box/1769-minimum-number-of-operations-to-move-all-balls-to-each-box.py
class Solution:
    def minOperations(self, bo: str) -> List[int]:
        m, b = 0, 0; ans = [0 for _ in range(len(bo))]

        for i in range(len(bo)):
            ans[i] = m + b
            m += b
            b += int(bo[i])

        m, b = 0, 0

        for i in reversed(range(len(bo))):
            ans[i] += m + b
            m += b
            b += int(bo[i])

        return 